### PR TITLE
[MIG] runbot_travis2docker: Migration from 9.0 to 11.0

### PR DESCRIPTION
**Pull request by moylop260**

### CODE REVIEW

**Comment by moylop260** [comment]
cc @lasley @pedrobaeza @jjscarafia @alan196

File: runbot_travis2docker/README.rst
Comment (hbrunn) [inline runbot_travis2docker/README.rst]: *functionality* (in both lines)

File: runbot_travis2docker/models/runbot_branch.py
Comment (hbrunn) [inline runbot_travis2docker/models/runbot_branch.py]: no short circuit if `uses_weblate` is false?

File: runbot_travis2docker/models/runbot_branch.py
Comment (hbrunn) [inline runbot_travis2docker/models/runbot_branch.py]: this and the following lines need one level of indentation more right?

File: runbot_travis2docker/models/runbot_branch.py
Comment (hbrunn) [inline runbot_travis2docker/models/runbot_branch.py]: I'd add those to the search, but doesn't matter so much I guess

File: runbot_travis2docker/models/runbot_build.py
Comment (hbrunn) [inline runbot_travis2docker/models/runbot_build.py:98]: why don't you call super here and patch `_spawn` such that is runs another command when it's a docker build and the command is what core would do here?

File: runbot_travis2docker/models/runbot_build.py
Comment (hbrunn) [inline runbot_travis2docker/models/runbot_build.py]: this merits a comment i think

File: runbot_travis2docker/models/runbot_build.py
Comment (hbrunn) [inline runbot_travis2docker/models/runbot_build.py]: please make this `_get_docker_run_cmd` (or similar, but that's very generic and bound to clash with other stuff i guess)

**Review by hbrunn — CHANGES_REQUESTED** [review]
I think there's only one breaking comment, the rest if good to have imho

Should we maybe adopt optimistic merging for this project in a similar style as we do in https://github.com/OCA/OpenUpgrade/issues/511 ? I think it's a good thing to really start merging stuff into 11, and if we wait until every comma is fixed, this will happen just as early as for the open PRs we have already now.

So if people agree, I'd merge this one, PR my nags in a new PR, if it's green, the next person stopping by presses the merge button

**Comment by hbrunn** [comment]
I do not understand btw why weblate stuff is intermingled with the docker module, is that connected in some way? Otherwise, I think it would be a better architecture to have hooks to support $translationservice, but well, see the stuff about optimistic merging above, when we need it, we'll PR is

File: .travis_requirements.sh
Comment (elicoidal) [inline .travis_requirements.sh]: I would prefer a OCA related docker

File: runbot_travis2docker/README.rst
Comment (elicoidal) [inline runbot_travis2docker/README.rst]: png

File: runbot_travis2docker/README.rst
Comment (elicoidal) [inline runbot_travis2docker/README.rst]: Enumerated list

File: runbot_travis2docker/README.rst
Comment (elicoidal) [inline runbot_travis2docker/README.rst]: runs

File: runbot_travis2docker/__init__.py
Comment (elicoidal) [inline runbot_travis2docker/__init__.py]: remove licenses from init.py

File: runbot_travis2docker/__manifest__.py
Comment (elicoidal) [inline runbot_travis2docker/__manifest__.py]: in README: contributors

File: runbot_travis2docker/__manifest__.py
Comment (elicoidal) [inline runbot_travis2docker/__manifest__.py]: target repo URL

File: runbot_travis2docker/__manifest__.py
Comment (elicoidal) [inline runbot_travis2docker/__manifest__.py]: space after the comma

File: runbot_travis2docker/__manifest__.py
Comment (elicoidal) [inline runbot_travis2docker/__manifest__.py:16]: single quotes

File: runbot_travis2docker/data/ir_cron_data.xml
Comment (elicoidal) [inline runbot_travis2docker/data/ir_cron_data.xml]: remove encoding

File: runbot_travis2docker/demo/runbot_repo_demo.xml
Comment (elicoidal) [inline runbot_travis2docker/demo/runbot_repo_demo.xml]: should not we have this in OCA?

File: runbot_travis2docker/models/runbot_branch.py
Comment (elicoidal) [inline runbot_travis2docker/models/runbot_branch.py:18]: strings?

File: runbot_travis2docker/models/runbot_branch.py
Comment (elicoidal) [inline runbot_travis2docker/models/runbot_branch.py]: use constant?

File: runbot_travis2docker/models/runbot_repo.py
Comment (elicoidal) [inline runbot_travis2docker/models/runbot_repo.py]: I would make this URL a setup or constant

File: runbot_travis2docker/models/runbot_repo.py
Comment (elicoidal) [inline runbot_travis2docker/models/runbot_repo.py]: Wrong JSON response format

File: runbot_travis2docker/models/runbot_repo.py
Comment (elicoidal) [inline runbot_travis2docker/models/runbot_repo.py]: weblate -> Weblate server

**Review by elicoidal — CHANGES_REQUESTED** [review]

File: runbot_travis2docker/__manifest__.py
Comment (moylop260) [reply runbot_travis2docker/__manifest__.py:16]: I have used double quotes like as our template:
 - https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py

File: runbot_travis2docker/models/runbot_branch.py
Comment (moylop260) [reply runbot_travis2docker/models/runbot_branch.py:18]: I'm using our standard of [fields](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#fields)

`If the technical name of the field (the variable name) is the same to the string of the label, don't put string parameter for new API fields, because it's automatically taken. If your variable name contains "_" in the name, they are converted to spaces when creating the automatic string and each word is capitalized...`

File: runbot_travis2docker/models/runbot_branch.py
Comment (moylop260) [reply runbot_travis2docker/models/runbot_branch.py]: I don't understand this comment.
What variable do you want a constant? (cmd, port, 22?)

File: .travis_requirements.sh
Comment (moylop260) [reply .travis_requirements.sh]: This PR is a [MIG] of current code from 9.0
Use an OCA image (That is not exists) is a big improvement not related to the focus of this PR (Migration)

File: runbot_travis2docker/models/runbot_branch.py
Comment (moylop260) [reply runbot_travis2docker/models/runbot_branch.py]: Good catch!
Thank you

File: runbot_travis2docker/models/runbot_branch.py
Comment (moylop260) [reply runbot_travis2docker/models/runbot_branch.py]: You are right!
A search is faster than a python validation.
Thanks!

File: runbot_travis2docker/__manifest__.py
Comment (moylop260) [reply runbot_travis2docker/__manifest__.py]: If you add a space after comma here you will have a PEP error here `trailing whitespace`
and our standard has not defined this change to the PEP
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms

`When making a comma separated list, dict, tuple, ... with one element per line, append a comma to the last element. This makes it so the next element added only changes one line in the changeset instead of changing the last element to simply add a comma.`

File: runbot_travis2docker/models/runbot_build.py
Comment (moylop260) [reply runbot_travis2docker/models/runbot_build.py]: Good point!
Thanks!

File: runbot_travis2docker/models/runbot_build.py
Comment (moylop260) [reply runbot_travis2docker/models/runbot_build.py:98]: This is a good idea but the main parameter `build` of this job is not received for the method `spawn`
This could be possible if odoo guys use `self` instead of `build` or send this parameter to `spawn` method.

In order to knows if the build is a travis2docker one.

**Comment by moylop260** [comment]
@elicoidal @hbrunn 
Thanks for your review...
All comments were fixed or commented.

**Comment by moylop260** [comment]
+1 to use Optimistic Merge for this project

File: runbot_travis2docker/models/__init__.py
Comment (elicoidal) [inline runbot_travis2docker/models/__init__.py:1]: no license in init.py

**Review by elicoidal — APPROVED** [review]
+1 for optimistic merging

**Review by hbrunn — APPROVED** [review]

File: runbot_travis2docker/models/__init__.py
Comment (moylop260) [reply runbot_travis2docker/models/__init__.py:1]: I'm using our template:
https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py#L2

Should we delete this part from the template?
Feel free of create a new PR to MT in order to delete it, please

**Pull request merged**
